mifospay/src/main/java/org/mifos/mobilewallet/mifospay/bank/fragment/DebitCardFragment.kt

```diff
@@ -21,102 +22,47 @@ import org.mifos.mobilewallet.mifospay.bank.presenter.DebitCardPresenter
-    var mDebitCardPresenter: BankContract.DebitCardPresenter? = null
-
-    @JvmField
-    @BindView(R.id.et_debit_card_number)
-    var mEtDebitCardNumber: EditText? = null
-
-    @JvmField
-    @BindView(R.id.pe_month)
-    var mPeMonth: PinEntryEditText? = null
-
-    @JvmField
-    @BindView(R.id.pe_year)
-    var mPeYear: PinEntryEditText? = null
-    var key = false
-
+class DebitCardFragment : BaseFragment() {
     override fun onCreateView(
         inflater: LayoutInflater, container: ViewGroup?,
         savedInstanceState: Bundle?
     ): View {
-        val rootView = inflater.inflate(
-            R.layout.fragment_debit_card,
-            container, false
-        ) as ViewGroup
-        ButterKnife.bind(this, rootView)
-        mPresenter!!.attachView(this)
-        mPeYear!!.setOnEditorActionListener(OnEditorActionListener { v, actionId, event ->
-            if (actionId == EditorInfo.IME_ACTION_DONE) {
-                okayClicked()
-                return@OnEditorActionListener true
-            }
-            false
-        })
-        mPeMonth!!.addTextChangedListener(object : TextWatcher {
-            override fun beforeTextChanged(s: CharSequence, start: Int, count: Int, after: Int) {}
-            override fun onTextChanged(s: CharSequence, start: Int, before: Int, count: Int) {
-                if (s.length == 2) {
-                    mPeYear!!.requestFocus()
-                }
-            }
 
-            override fun afterTextChanged(s: Editable) {}
-        })
-        mPeYear!!.setOnKeyListener(View.OnKeyListener { v, keyCode, event ->
-            if (keyCode == KeyEvent.KEYCODE_DEL && mPeYear!!.length() == 0) {
-                if (key) {
-                    mPeMonth!!.requestFocus()
-                    mPeMonth!!.dispatchKeyEvent(
-                        KeyEvent(KeyEvent.ACTION_DOWN, KeyEvent.KEYCODE_DEL)
+        return ComposeView(requireContext()).apply {
+            setContent {
+                MifosTheme {
+                    DebitCardScreen(
+                        onDebitCardVerified = { otp ->
+                            verifyDebitCardSuccess(otp)
+                        },
+                        onDebitCardVerificationFailed = { message ->
+                            verifyDebitCardError(message)
+                        }
                     )
-                    key = false
-                    return@OnKeyListener false
                 }
-                key = true
             }
-            false
-        })
-        return rootView
-    }
-
-    private fun okayClicked() {
-        showProgressDialog(Constants.PLEASE_WAIT)
-        mDebitCardPresenter!!.verifyDebitCard(mEtDebitCardNumber!!.text
-            .toString().trim { it <= ' ' },
-            mPeMonth!!.text.toString(),
-            mPeYear!!.text.toString()
-        )
+        }
     }
 
-    override fun verifyDebitCardSuccess(otp: String?) {
+     fun verifyDebitCardSuccess(otp: String?) {
```

Comment:
move this success and error state in compose function

Reply:
done